scripts/archive_api.py

```diff
@@ -77,6 +107,57 @@ async def health_check():
 	        "HopAuthOK": hop_auth_ok,
 	       }
 
+messageRecordSchema = {
+                          "type": "object",
+                          "properties": {
+                              "message": {
+                                  "type": "string",
+                                  "format": "binary",
+                                  "description": "The raw mesage body"
+                              },
+                              "metadata": {
+                                  "type": "object",
+                                  "description": "Kafka metadata for the message",
+                                  "properties": {
+                                      "timestamp": {
+                                          "type": "number",
+                                          "description": "The time at which the message was sent to the Kafka broker"
+                                      },
+                                      "headers": {
+                                          "type": "object",
+                                          "description": "Kafka message headers",
+                                          "patternProperties": {
+                                              ".*": {
+                                                  "type": "string",
+                                                  "format": "binary",
+                                              }
+                                          }
+                                      },
+                                      "key": {
+                                          "type": "string",
+                                          "format": "binary",
+                                          "description": "Kafka message key"
+                                      },
+                                  }
+                              },
+                              "annotations": {
+                                  "type": "object",
+                                  "description": "Metadata added by the archive",
+                                  "properties": {
+                                      "con_message_crc32": {
+                                          "type": "number",
+                                          "description": "The CRC32 of the message data"
+                                      },
+                                      "con_text_uuid": {
+                                          "type": "string",
+                                          "format": "uuid",
+                                          "description": "The message UUID as a string"
```

Comment:
How is this different from the headers `_id`? I'm using that in hermes for the uuid so I hoping I'm not doing it wrong...

Reply:
This is different in that:
1. It is an output from the system rather than an input from the user
2. It may not have the same value as what the user set in the `_id` header, since it is entirely possible for the user to accidentally or deliberately re-use a UUID. Internally, the archive checks for duplicates, and re-assigns the ID of an incoming message if it collides with one stored previously. 

Something on my to-do list is to extend the upload response to contain the final ID of the message, both so that the client can know if it had to be reassigned, and so that a client can be lazy and let the server make the assignment.

Reply:
Ah okay, that would be super useful for us so we don't generate the wrong url to download uploaded files!